### PR TITLE
Introduce concepts in documentation

### LINKED ISSUE #228: Better separation of Documentation and Concepts
## General

* pyfar version: 0.2.x

## Description

There are parts of the documentation that pertain to concepts of pyfar rather than documenting functions or classes. Examples are:
- [x]  Concept of Energy and Power signals (Simon)
- [x]  Explanation of coordinates systems (Fabian)
- [x]  Explanations of FFT related stuff (Simon, Alex)
- [x]  Explanations of signal arithmetics (Simon)
- [x]  Filter types (Fabian)

A possibility to speparate might be to add a section 'Concepts' to the docs. Parts of these section can be linked from the *real* documentation without loosing information and making the concepts easier to access at the same time.

### PR DESCRIPTION
**Pull request by sikersten**
### Which issue(s) are closed by this pull request?

Closes #228 

### Changes proposed in this pull request:

First version of concepts in documentation
- added concepts
- new folder structure in docs
- extracted concepts for audio, filter, coordinates, fft in ..._concepts.py files
- separated concepts for audio classes and arithmetic operations

As there are a lot of file changes due to the restructuring, build the documentation locally to inspect it.

So far, no new information was added. We can discuss, if we would also add information in the new concept files as part of this PR.

Unfortunately, I messed up with the develop branch and pull request #102, sorry!

### CODE REVIEW

**Review by al-de — APPROVED** [review]
Thanks a lot for restructuring the documentation. For me this looks fine. :)

File: docs/concepts.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Concepts
+========
+
```
Comment (f-brinkmann) [inline docs/concepts.rst:3]:
```suggestion

The sections below give background information that is helpful to fully
understand how pyfar works and to understand how to work with pyfar. If you
are new to pyfar and or acoustics signal processing, we recommend briefly going
through these concepts.

```

File: docs/classes.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ Classes
 .. toctree::
    :maxdepth: 1
```
Comment (f-brinkmann) [inline docs/classes.rst:6]:
```suggestion

pyfar uses classes for convenient handling of certain audio and meta data.
These classes are introduced below.

```

File: docs/modules.rst [hunk 3]
```diff
@@ -4,10 +4,10 @@ Modules
 .. toctree::
    :maxdepth: 1
```
Comment (f-brinkmann) [inline docs/modules.rst:6]:
```suggestion

The following gives detailed information about all pyfar functions sorted
according to their modules.

```

**Review by f-brinkmann — APPROVED** [review]
Thanks - I think this really makes it better for users to start with pyfar and for us to maintain the docs. I only suggested to add some additional information to some files ... and I deleted a deprecated to do file in a single commit.

**Pull request merged**